hashmap.js

```diff
@@ -25,7 +25,14 @@
 		this.clear();
 		switch (arguments.length) {
 			case 0: break;
-			case 1: this.copy(other); break;
+			case 1: {
+				if(Array.isArray(other)) {

```

Comment:
- nitpick: space after the `if`
- change `Array.isArray(other)` for `'length' in other`